### ISSUE TITLE
Default style is defined as an empty string in standard and QGIS 3 - Lizmap and QGIS has to be fixed

### ISSUE THREAD

**Issue by 80ben** [3liz/lizmap-web-client#1119]
I have a problem with the Legend icon in LIZMAP with  lizmap Plugin 3.0.1 in QGIS 3.6. The layer symbol is not displayed. This is url’s image

http://www.......   @=2&project=fatto&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetLegendGraphic&LAYER=postattivazioni&STYLE=default&EXCEPTIONS=application%2Fvnd.ogc.se_inimage&FORMAT=image%2Fpng&TRANSPARENT=TRUE&WIDTH=150&LAYERFONTSIZE=0&ITEMFONTSIZE=9&SYMBOLSPACE=1&ICONLABELSPACE=2&DPI=96&LAYERSPACE=0&LAYERFONTBOLD=FALSE&LAYERTITLE=FALSE

I use Ubuntu 18.04.2 LTS, qgis-server 3.6, qgis 3.6, Lizmap web client 3.2.2, postgresql 10.6, postgis 2.4.3 python 2.7.11 & 3.6.7 and php 7.2

In my opinion, there is a bug in  lizmap Qgis Python Plugin ( > v 3.00), because  in previous the versions the this bug didn’t exist! maybe the cfg.file is configured  bad.

![immagine](https://user-images.githubusercontent.com/30823307/53960822-897ffc00-40e7-11e9-87b3-820a8f02d0f2.png)

@kimaidou

**Comment by rldhont** [comment]
It is not a lizmap-plugin issue.

Lizmap plugin is a tool to configure the web map. Lizmap Web Client build the Web User Interface and it is a proxy to QGIS Server.

Lizmap Web Client 3.2.2 is not fully functional with QGIS Server 3.*

And it seems, it is a QGIS Server issue in the GetLegendGraphic Request.

Can you test the request directly to QGIS Server ?
* Replace the lizmap url to the QGIS Server once (accessible from your browser)
* Remove repository and project parameters from the URL
* Add the map parameter with the full path to the project

Can you take a look at the lizmap log and the QGIS Server log.

**Comment by 80ben** [comment]
I solved using Qgis -Dessktop 2.14 "Essen" to configure qgs and cfg files.


Il 20/03/2019 12:00, rldhont ha scritto:
> Can you take a look at the lizmap log and the QGIS Server log

-- 
dott. ing. Benedetto Figorito
Coll. Tecnico Prof.le - Esperto GIS
ARPA PUGLIA - Direzione Scientifica - U.O.S. Agenti Fisici
C.so Trieste n. 27
70126 – Bari
Tel.: 080-5460.312 (diretto)
E-mail: b.figorito@arpa.puglia.it
Sito web ARPA: www.arpa.puglia.it
WebGis: www.webgis.arpa.puglia.it

**Comment by rldhont** [comment]
@80ben ok, but you probably find a bug in QGIS Server and I'd like to fix it

**Comment by Abelon38** [comment]
Hi!

I have the same issue and it does work when creating and saving the project with QGIS 2.18. It would be womderful if everything works with QGIS 3.6. My environment looks like this:

- Tested on windows 10 and windows server 2012
- QGIS server 3.6
- Lizmap 3.3 and 3.3 pre
- Phyton 37
- QGIS server installed with Xampp, according QGIS docs.
Error from Lizamp: 2019-03-22 11:17:28	::1	warning	2019-03-22 11:17:28	[2]	SQLite3::query(): Unable to prepare statement: 1, no such table: lizmap_search	C:\lizmap\lib\jelix\plugins\db\sqlite3\sqlite3.dbconnection.php	112

2019-03-22 11:17:29	::1	error	 error key C:\qgis-projekt/hasseholm/hasseholm.qgs, use only alphanumeric characters and the character '_' and maximum length 255
2019-03-22 11:17:29	::1	error	 error key C:\qgis-projekt/hasseholm/hasseholm.qgs, use only alphanumeric characters and the character '_' and maximum length 255

Thanks in advaced!
Abel

**Comment by rmarzocchi84** [comment]
Same issue which seems depend on QGIS server (the problem affect also print function)

Ubuntu 18.04
QGIS Server 3.4.6



http://   MYHOST/cgi-bin/qgis_mapserv.fcgi?map=/xxxx/xxxx/xxxxx/nextcloud-data/workshop/files/progetto_roberto_marzocchi.qgs&LAYERS=palazzi_rolli&STYLES=default&VERSION=1.3.0&EXCEPTIONS=application%2Fvnd.ogc.se_inimage&FORMAT=image%2Fpng&DPI=96&TRANSPARENT=TRUE&SERVICE=WMS&REQUEST=GetMap&CRS=EPSG%3A3857&BBOX=992344.5248270923,5528401.960096031,996118.6030981477,5530054.910832468&WIDTH=1580&HEIGHT=692


```
<ServiceExceptionReport xmlns="http://www.opengis.net/ogc" version="1.3.0">
<ServiceException code="StyleNotDefined">
Style "default" does not exist for layer "palazzi_rolli"
</ServiceException>
</ServiceExceptionReport>
```

It is better to open an issue also on QGIS redmine platform (https://issues.qgis.org/projects/qgis/issues/)?

**Comment by rmarzocchi84** [comment]
I probably understand the problem.. QGIS server automatic translate the "default" style. In my case is translated in Italian "predefinito" 

So if I test the WMS with this parameter it works correctly:

http:// MYHOST/cgi-bin/qgis_mapserv.fcgi?map=/xxxx/xxxx/xxxxx/nextcloud-data/workshop/files/progetto_roberto_marzocchi.qgs&LAYERS=palazzi_rolli&STYLES=**predefinito**&VERSION=1.3.0&EXCEPTIONS=application%2Fvnd.ogc.se_inimage&FORMAT=image%2Fpng&DPI=96&TRANSPARENT=TRUE&SERVICE=WMS&REQUEST=GetMap&CRS=EPSG%3A3857&BBOX=992344.5248270923,5528401.960096031,996118.6030981477,5530054.910832468&WIDTH=1580&HEIGHT=692

**Comment by rmarzocchi84** [comment]
@rldhont I see you have change the title of issue and you are right!

From my test I understand (perhaps is obvious for you and sorry for the noise, but maybe can be useful):

- old  project (QGIS 2):  
```
      <map-layer-style-manager current="">
        <map-layer-style name=""/>
      </map-layer-style-manager>
```

- new project (QGIS 3) in italian version but i think the same problem there is using not english language

```
      <map-layer-style-manager current="predefinito">
        <map-layer-style name="predefinito">
       [....]
      </map-layer-style-manager>
```

In fact changing the QGIS project in this way solve the problem (but obviously is not user friendly for lizmap users): 

```
      <map-layer-style-manager current="default">
         <map-layer-style name="default">
         <map-layer-style name="predefinito">
       [....]
      </map-layer-style-manager>
```

**Comment by rldhont** [comment]
Thanks you @rmarzocchi84 to identify the issue! We know the workaround and would like to avoid it. And we found an issue in QGIS Server.

**Comment by rldhont** [comment]
Working on a fix to QGIS server

**Comment by rldhont** [comment]
Fix for QGIS Server provided https://github.com/qgis/QGIS/pull/9696

**Comment by pcav** [comment]
In my case the "fix" is simply replacing 'predefinito' with 'default'. I imagine using qgis in English would avoid the issue, pending the new version, right?
Thanks a lot.

**Comment by rldhont** [comment]
@pcav you're right. The issue is that QGIS 3 stored default style with its translated name.

In QGIS 2, the default style has an empty name, replaced by default in QGIS Server. The solution is not to forced default but that QGIS Server keeps empty style name in the WMS request.

**Comment by rmarzocchi84** [comment]
@pcav i confirm the problem/solution of translation,  but, if i correctly understand the discussion in the qgis repository, the "default" parameter is not a WMS standard

**Comment by rldhont** [comment]
@rmarzocchi84 you are well understood 'default' is not a WMS standard for default style, it's an empty string.

To fix Lizmap, I have prepared some codes:
* Use the explicit name of style
* If no style name provided for groups use default for 2.18 or empty string for 3.* but there is an issue in printing in this last case.

So the fix will be fully functional for QGIS 3.4.7
